bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,286 @@
+
+; Transaction Input
+: [TC] aka Input. An element of the input list of the transaction. Must include an outpoint, an input script, and a sequence. The input script can have various content, corresponding to the UTXO that is being spent. The input script has length 0 for native segwit inputs. In segwit transactions, every input must be accompanied by a Witness Stack. The Witness Stack is not part of the Transaction Input.
+
+; Transaction Input List
+: [AT] The enumeration of all Transaction Inputs of a transaction.
+
+; Transaction Output
+: [TC] Creates a UTXO by specifying an amount and an output script.
+
+; Transaction Output List
+: [AT] The enumeration of all Transaction Outputs of a transaction
+
+; txid
+: [TC] Part of an Outpoint that identifies the transaction that created a UTXO. Result of hashing a transaction without its witness structure.
+
+; Version
+: [TC] Part of the transaction header that specifies the transaction version.  Serializes as the four byte <code>nVersion</code> integer field.
+
+; Witness Item
+: [TC] An element of a witness stack. Witness Items are: Witness Script (P2WSH only), Script Arguments, Leaf Script (P2TR scriptpath only), Control Block (P2TR scriptpath only), Annex (P2TR only). BIP341 also refers to Witness Items as “Witness Elements” or “Witness Stack Elements”.
+
+; Witness Program
+: [AT] A script template that forwards input validation to the Witness Stack. Witness Programs are a type of Forwarding Script. Witness Programs appear in the Output Script for native segwit outputs and in the Input Script for wrapped segwit inputs.
+
+; Witness Structure
+: [TC] The part of the serialized transaction that contains the witness stacks for each input.
+
+; Witness Script
+: [TC] A Witness Item that provides the Condition Script used to spend P2WSH outputs.
+
+; Witness Stack
+: [TC] The pre-built stack of witness items that is executed in validation to satisfy an input's condition script. If a transaction has at least one segwit input, each input has exactly one witness stack.  Witness stacks are not scripts, but may include witness items that are scripts. While there is a witness stack for each input, the witness stack is not part of the input. Strictly speaking, a witness stack can contain witness items that will not be added to the stack, such as the Control Block.
+
+; Witness Version
+: [TC] The version of the witness program in native segwit outputs.
+
+; wtxid
+: [AT] Witness Transaction Identifier. Result of hashing a complete transaction including its witness structure. Used in transaction propagation and to construct the Witness Commitment.
+
+===Usage in context of UTXO creation and spending===
```

Comment:
The table in this section is GREAT :raised_hands:, really handy and I don't think I've come across something like it before.